util/quarantine/quarantine_darwin.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package quarantine
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"golang.org/x/sys/unix"
+)
+
+func setQuarantineAttr(f *os.File) error {
+	sc, err := f.SyscallConn()
+	if err != nil {
+		return err
+	}
+
+	now := time.Now()
+
+	// We uppercase the UUID to match what other applications on macOS do
+	id := strings.ToUpper(uuid.New().String())
+
+	// This format is under-documented, but the following links contain a
+	// reasonably comprehensive overview:
+	//    https://eclecticlight.co/2020/10/29/quarantine-and-the-quarantine-flag/
+	//    https://nixhacker.com/security-protection-in-macos-1/
+	//    https://ilostmynotes.blogspot.com/2012/06/gatekeeper-xprotect-and-quarantine.html
+	attrData := fmt.Sprintf("%s;%x;%s;%s",
+		"0002",      // quarantine value; kLSQuarantineTypeOtherDownload
```

Comment:
Per https://nixhacker.com/security-protection-in-macos-1/ `kLSQuarantineTypeOtherDownload` is 1 not 2. I also see a leading 8 most of the time, which I think implies that this is a bitfield that has other values too. 

https://github.com/keybase/client/blob/457ebc8fc401c44dccf2ce684267b802fe61ea74/go/kbfs/libfuse/quarantine_darwin.go#L26-L28 has `_kLSQuarantineTypeOtherDownload = "0081;"`, which would agree with that.

Though if I use the AppKit APIs (`NSURL`'s `setResourceValue(..., forKey: kCFURLQuarantinePropertiesKey as URLResourceKey)`) with `kLSQuarantineTypeKey` set to `kLSQuarantineTypeOtherDownload` I end up with `0083`, so 🤷

Reply:
Whoops, off-by-one 🤦‍♂️  Switched to `1` and verified that this still works:
```
$ xattr -p com.apple.quarantine ./quarantine-test
0001;63766eee;Tailscale;b3e2ed8b-5e8d-4774-ba7e-9e6ff89adb06
```

<img width="372" alt="Screenshot 2022-11-17 at 2 56 35 PM" src="https://user-images.githubusercontent.com/1079173/202546213-9c990553-93d8-4381-a361-f1d402895b17.png">